### ISSUE TITLE
Incorrect warning:  SC2089 (warning): Quotes/backslashes will be treated literally. Use an array.

### ISSUE THREAD

**Issue by vipasko** [koalaman/shellcheck#2861]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2089, SC2090, SC2086, SC2089, SC2090, SC2086
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash

LC_CTYPE=C.UTF-8

echo
echo small-alpha
alpha='ɑ' # latin small letter alpha
echo -n "$alpha" | xxd -p # expected UTF-8 c991
echo -n "$alpha" | xxd -p | xxd -p -r # result ɑ
echo -en '\xC9\x91' # result ɑ # latin small letter alpha
printf '\xC9\x91' # result ɑ # latin small letter alpha
echo
printf -v alpha_4xchar '%04x' "'$alpha"
echo $alpha_4xchar # result Unicode 0251
printf -v alpha_2xchar '%02x' "'$alpha"
echo $alpha_2xchar # result Unicode 251 <= why 251 but format is %02x
printf '%02x' "'$alpha"
/bin/printf 
```

#### Here's what shellcheck currently says:
```
Line 13	SC2089: Quotes/backslashes will be treated literally. Use an array.
Line 14	SC2090: Quotes/backslashes in this variable will not be respected.
Line 14	SC2086: Double quote to prevent globbing and word splitting.
Line 15	SC2089: Quotes/backslashes will be treated literally. Use an array.
Line 16	SC2090: Quotes/backslashes in this variable will not be respected.
Line 16	SC2086: Double quote to prevent globbing and word splitting.
```
#### Here's what I wanted or expected to see:

TODO: Describe expected/desired output

\U0251'
printf 
```

#### Here's what shellcheck currently says:
```
%ACTUAL%
```
#### Here's what I wanted or expected to see:

TODO: Describe expected/desired output

\U0251'
echo -en '\U0251' # to get 'ɑ' # latin small letter alpha
```

#### Here's what shellcheck currently says:
```
%ACTUAL%
```
#### Here's what I wanted or expected to see:

TODO: Describe expected/desired output

**Comment by GfEW** [comment]
I think the repeated occurrences of `TODO: Describe expected/desired output` make this issue difficult to understand, let alone resolve.

**Comment by brother** [comment]
I lean towards that the SC2089 warning on line 13 and 15 is correct. Someone could probably make the case that shellcheck should know how `printf` and the single quote character works and then not warn about it - I am not sure about that.
Ignoring the warning with an directive (`# shellcheck disable=SC2089`)  seems appropriate in this case. 

For SC2086 and SC2090 at line 13 and 16 are a different story. I am on the camp for always quoting when using echo and thus silence them, but people are on the fence over that =)
